### ISSUE TITLE
install_github is failing on a large package

### ISSUE THREAD

**Issue by boshek** [r-lib/remotes#262]
tl;dr remotes not installing large data package from github where older version of devtools was

My issue is that I am trying to install a bigger data package we've created to hold a bunch of spatial objects.  When I try to install with the CRAN devtools/remotes I get this error:

``` r
if("bcmaps.rdata" %in% installed.packages()) remove.packages("bcmaps.rdata")
#> Removing package from 'C:/Users/salbers/R/win-library/3.5'
#> (as 'lib' is unspecified)
if("remotes" %in% installed.packages()) remove.packages("remotes")
#> Removing package from 'C:/Users/salbers/R/win-library/3.5'
#> (as 'lib' is unspecified)

install.packages("remotes")
#> Installing package into 'C:/Users/salbers/R/win-library/3.5'
#> (as 'lib' is unspecified)
#> package 'remotes' successfully unpacked and MD5 sums checked
#> 
#> The downloaded binary packages are in
#>  C:\Users\salbers\AppData\Local\Temp\Rtmpcjdtsf\downloaded_packages

library(remotes)
install_github("bcgov/bcmaps.rdata")
#> Downloading GitHub repo bcgov/bcmaps.rdata@master
#>   

   checking for file 'C:\Users\salbers\AppData\Local\Temp\Rtmpcjdtsf\remotes562427122e46\bcgov-bcmaps.rdata-78fdaff/DESCRIPTION' ...
     checking for file 'C:\Users\salbers\AppData\Local\Temp\Rtmpcjdtsf\remotes562427122e46\bcgov-bcmaps.rdata-78fdaff/DESCRIPTION' ... 
  v  checking for file 'C:\Users\salbers\AppData\Local\Temp\Rtmpcjdtsf\remotes562427122e46\bcgov-bcmaps.rdata-78fdaff/DESCRIPTION' (393ms)
#>   
-  preparing 'bcmaps.rdata': (11.5s)
#> 
    checking DESCRIPTION meta-information ...
     checking DESCRIPTION meta-information ... 
 v  checking DESCRIPTION meta-information
#>  
-  checking for LF line-endings in source and make files and shell scripts (965ms)
#>   
-  checking for empty or unneeded directories
#>   
-  looking to see if a 'data/datalist' file should be added
#>   
Warning in gzfile(file, "rb") :
    Warning in gzfile(file, "rb") :
#>      cannot open compressed file 'bcmaps.rdata', probable reason 'Permission denied'
#>    Error in gzfile(file, "rb") : cannot open the connection
#>    Execution halted
#> 
#> Error in run(bin, args = real_cmdargs, stdout_line_callback = real_callback(stdout), : System command error

packageVersion("bcmaps.rdata")
#> Error in packageVersion("bcmaps.rdata"): package 'bcmaps.rdata' not found
packageVersion("remotes")
#> [1] '2.0.2'
```

<sup>Created on 2018-11-27 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

Other github based installs proceed fine. So my thought was let's try this on a previous CRAN version of `devtools` (prior to fully splitting installation off to `remotes`). So I made a reprex:

```
if("bcmaps.rdata" %in% installed.packages()) remove.packages("bcmaps.rdata")
if("devtools" %in% installed.packages()) remove.packages("devtools")


packageurl <- "https://cran.r-project.org/src/contrib/Archive/devtools/devtools_1.13.6.tar.gz"
install.packages(packageurl, repos=NULL, type="source")
#> Installing package into 'C:/Users/salbers/R/win-library/3.5'
#> (as 'lib' is unspecified)

library(devtools)
install_github("bcgov/bcmaps.rdata")
#> Using GitHub PAT from envvar GITHUB_PAT
#> Downloading GitHub repo bcgov/bcmaps.rdata@master
#> from URL https://api.github.com/repos/bcgov/bcmaps.rdata/zipball/master
#> Installing bcmaps.rdata
#> "C:/PROGRA~1/R/R-35~1.1/bin/x64/R" --no-site-file --no-environ --no-save  \
#>   --no-restore --quiet CMD INSTALL  \
#>   "C:/Users/salbers/AppData/Local/Temp/RtmpMl6cmg/devtools4368419a666/bcgov-bcmaps.rdata-78fdaff"  \
#>   --library="C:/Users/salbers/R/win-library/3.5" --install-tests
#> 

packageVersion("bcmaps.rdata")
#> [1] '0.2.0'
packageVersion("devtools")
#> [1] '1.13.6'
```
<sup>Created on 2018-11-27 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

So with the older version of `devtools` (1.13.6), large downloads works fine. But on version 2.0.2 of remotes I am unable to download. 

Here is my session info though it is a little inaccurate because I am downloading new packages in this issue:

<details> <summary><strong>Session Info</strong></summary>

```r
> sessionInfo()
R version 3.5.1 (2018-07-02)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 17134)

Matrix products: default

locale:
[1] LC_COLLATE=English_Canada.1252  LC_CTYPE=English_Canada.1252    LC_MONETARY=English_Canada.1252 LC_NUMERIC=C                   
[5] LC_TIME=English_Canada.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
[1] compiler_3.5.1 tools_3.5.1   
```
</details>

**Comment by gaborcsardi** [comment]
This has not much to do with package sizes, but `R CMD build` throws an warning, and remotes errors out on warnings, because they often lead to broken installations.

You can tell remotes to ignore warnings, see `R_REMOTES_NO_ERRORS_FROM_WARNINGS` at https://github.com/r-lib/remotes#environment-variables, or you can try to install the package without calling `R CMD build` on it, that's what devtools used to do:
```r
remotes::install_github(..., build = FALSE)
```

**Comment by jimhester** [comment]
I think the root issue is something different, the warning issued is coming from the build rather than the install (only warnings during installation are converted to errors) and the install succeeds despite the warning on my macOS machine.

```r
remotes::install_github("git@github.com:bcgov/bcmaps.rdata.git")
#> Downloading GitHub repo bcgov/bcmaps.rdata@master
#> ✔  checking for file ‘/private/var/folders/dt/r5s12t392tb5sk181j3gs4zw0000gn/T/Rtmp5xZ1gX/remotescdb655974df2/bcgov-bcmaps.rdata-78fdaff/DESCRIPTION’ ...
#> ─  preparing ‘bcmaps.rdata’: (582ms)
#> ✔  checking DESCRIPTION meta-information ...
#> ─  checking for LF line-endings in source and make files and shell scripts
#> ─  checking for empty or unneeded directories
#> ─  looking to see if a ‘data/datalist’ file should be added
#>    Warning in doTryCatch(return(expr), name, parentenv, handler) :
#>      error reading the file
#> ─  building ‘bcmaps.rdata_0.2.0.tar.gz’
#> 
#> Installing package into ‘/Users/jhester/Library/R/3.5/library’
#> (as ‘lib’ is unspecified)
#> * installing *source* package ‘bcmaps.rdata’ ...
#> ** R
#> ** data
#> *** moving datasets to lazyload DB
#> ** byte-compile and prepare package for lazy loading
#> ** help
#> *** installing help indices
#> ** building package indices
#> ** testing if installed package can be loaded
#> * DONE (bcmaps.rdata)
#> Adding ‘bcmaps.rdata_0.2.0.tgz’ to the cache
```

**Comment by gaborcsardi** [comment]
Right, forget about the env var then!

However, on Windows `R CMD build`, does fail, and `build = FALSE` is a good workaround. 

@boshek you might want to check why the build fails.

But anyway, this is not a remotes issue, so I am closing it now.
